errors/error.go

```diff
@@ -94,6 +82,29 @@ func New(ctx context.Context, args ...interface{}) error {
 	return e
 }
 
+// WithContext sets the context of the error. However, it does not override the existing context if it has already been
+// set. In the cases where there is a child node, the child node's context would have already been propagated to the
+// current context.
+func (e *Error) WithContext(ctx context.Context) *Error {
+	if e.ctx == nil {
+		e.ctx = map[string]any{}
+	}
+
+	basics := common.GetBasics(ctx)
+	if basics != nil {
+		collection.Copy(basics, e.basics())
+		e.setBasics(basics)
+	}
+
+	extras := common.GetExtras(ctx)
+	if extras != nil {
+		collection.Copy(extras, e.extras())
+		e.setExtras(extras)
+	}
+
+	return e
+}
+
```

Comment:
@yanyi-wego Shall we add one more extra func, then don't need to do the type check every time we call `WithContext` cc @viet-wego  @aaron-wego 
```
WithContext(ctx context.Context, err goErrs.Error) (goErrs.Error) {
   // if type of err is  Error
   // copy every thing in the context  use WithContext
   // else create a new Error with the context?
}
```

Reply:
Aha, can ignore this